usrsctplib/netinet/sctp_callout.h

```diff
@@ -99,20 +87,26 @@ typedef struct sctp_callout sctp_os_timer_t;
 
 #define	SCTP_CALLOUT_ACTIVE	0x0002	/* callout is currently active */
 #define	SCTP_CALLOUT_PENDING	0x0004	/* callout is waiting for timeout */
-
-void sctp_os_timer_init(sctp_os_timer_t *tmr);
+#define	SCTP_CALLOUT_CANCELLED	0x0008	/* callout is cancelled */
+
+void sctp_os_timer_describe(const sctp_os_timer_t*, size_t, char*);
+int sctp_os_timer_compare(const sctp_os_timer_t*, const sctp_os_timer_t*);
+int sctp_os_timer_is_pending(const sctp_os_timer_t*);
+int sctp_os_timer_is_active(const sctp_os_timer_t*);
+void sctp_os_timer_deactivate(sctp_os_timer_t*);
+void sctp_os_timer_init(sctp_os_timer_t*);
 void sctp_os_timer_start(sctp_os_timer_t *, uint32_t, void (*)(void *), void *);
+int sctp_os_timer_cancel(sctp_os_timer_t*);
 int sctp_os_timer_stop(sctp_os_timer_t *);
 void sctp_handle_tick(uint32_t);
 
 #define SCTP_OS_TIMER_INIT	sctp_os_timer_init
 #define SCTP_OS_TIMER_START	sctp_os_timer_start
-#define SCTP_OS_TIMER_STOP	sctp_os_timer_stop
-/* MT FIXME: Is the following correct? */
-#define SCTP_OS_TIMER_STOP_DRAIN SCTP_OS_TIMER_STOP
-#define	SCTP_OS_TIMER_PENDING(tmr) ((tmr)->c_flags & SCTP_CALLOUT_PENDING)
```

Comment:
Note for reviewers:
I've found out that `c_flags` was read and mutated inconsistently: it mostly mutated under mutex `&SCTP_BASE_VAR(timer_mtx)`, but was read in many places directly. I believe this could be a reason of some timer related bugs.
I've decided to move reading `c_flags` under mutex and place it into dedicated function inside `sctp_callout.c` file.

#### The places where `c_flags` was read:
* ##### NOT holding `&SCTP_BASE_VAR(timer_mtx)` mutex:
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_indata.c#L2598
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_indata.c#L2629
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_indata.c#L2656
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_indata.c#L4299
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_indata.c#L4302
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_indata.c#L5201
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_indata.c#L5211
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_indata.c#L5216
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctputil.c#L1707
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctputil.c#L1747
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctputil.c#L1753
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctputil.c#L1759
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctputil.c#L2352
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_output.c#L5113
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_output.c#L8962
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_output.c#L9264
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_output.c#L9348
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_output.c#L10355
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_output.c#L10519
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_output.c#L10604
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet6/sctp6_usrreq.c#L393
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_usrreq.c#L394
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_usrreq.c#L6186
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_usrreq.c#L6216
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_usrreq.c#L6334
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_output.c#L6368
* ##### Holding `&SCTP_BASE_VAR(timer_mtx)` mutex:
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_callout.c#L133
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_callout.c#L170
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_callout.c#L171
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_callout.c#L211
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_callout.c#L238

#### The places where `c_flags` was written:
* ##### NOT holding `&SCTP_BASE_VAR(timer_mtx)` mutex:
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctputil.c#L1759
* ##### holding mutex `&SCTP_BASE_VAR(timer_mtx)`
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_callout.c#L154
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_callout.c#L171
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_callout.c#L211
https://github.com/sctplab/usrsctp/blob/913de3599feded8322882bdae69f346da5a258fc/usrsctplib/netinet/sctp_callout.c#L238